### ISSUE TITLE
Adjust expected errors for missing functions in `polyglot/meta/table.js`

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#5617]
In the `./test/rql_test/test-runner polyglot/meta/table.js` tests we check to see that a few methods are no longer defined on `r`, specifically `r.wait` (line 258), `r.rebalance()` (line 263), and `r.wait` with a configuration (line 348). Unfortunately the specific error message generated by the lack of method there seems to vary with the version of node we test with:
node 0.10:
```
Object function () {
  var args;
  args = 1 <= arguments.length ? slice.call(arguments, 0) : [];
  return rethinkdb.expr.apply(rethinkdb, args);
} has no method 'wait'
```
node 0.12: `TypeError: undefined is not a function`
node 4.2: `TypeError: r.wait is not a function`

Our testing system does not currently distinguish between Node 0.10 and Node 0.12, so even though we could do a slightly better job by isolating those, I think we should just look for the `TypeError` and not try and distinguish the message.

**Comment by larkost** [comment]
Anyone object to me removing the message check from this test? This would mean that it would only check that we get a `TypeError`, not what the message is.

**Comment by danielmewes** [comment]
@larkost Sounds fine to me.

**Comment by larkost** [comment]
The fix has been applied to `next` with 44eb8b5 and `v2.3.x` with 88ac25d.
